core/src/main/java/org/elasticsearch/cluster/DiskUsage.java

```diff
@@ -44,6 +51,40 @@ public DiskUsage(String nodeId, String nodeName, String path, long totalBytes, l
         this.path = path;
     }
 
+    public DiskUsage(StreamInput in) throws IOException {
+        this.nodeId = in.readString();
+        this.nodeName = in.readString();
+        this.path = in.readString();
+        this.totalBytes = in.readVLong();
+        this.freeBytes = in.readVLong();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(this.nodeId);
+        out.writeString(this.nodeName);
+        out.writeString(this.path);
+        out.writeVLong(this.totalBytes);
+        out.writeVLong(this.freeBytes);
+    }
+
+    public XContentBuilder toShortXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field("path", this.path);
+        builder.byteSizeField("total_bytes", "total", this.totalBytes);
+        builder.byteSizeField("used_bytes", "used", this.getUsedBytes());
+        builder.byteSizeField("free_bytes", "free", this.freeBytes);
+        builder.field("free_disk_percent", this.getFreeDiskAsPercentage());
+        builder.field("used_disk_percent", this.getUsedDiskAsPercentage());
```

Comment:
I'm good with one decimal point with the caveat that this endpoint really should not be being indexed.